File: gatorlight.cpp
/**
* Andy England @ SparkFun Electronics
* September 6, 2018
* https://github.com/sparkfun/pxt-light-bit
*
* Development environment specifics:
* Written in Microsoft PXT
* Tested with a SparkFun temt6000 sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/
#include "pxt.h"
#include <cstdint>
#include <math.h>

using namespace pxt;

int xPos = 0;
int yPos = 0;

namespace gatorlight {
    /*
    * Calculates the light in Lux based on the ADC value passed in. 1 step in adcVal is equal to .488 uA or .976 lux at 5V
    */
    //%
    uint16_t getLux() {
        int bla = 2;
        return bla;
    }

    uint16_t getXPos() {
        xPos = xPos + 1;
        if(xPos == 5)
        {
            xPos = 0
        }
        return xPos;
    }

    uint16_t getYPos() {
        yPos = yPos + 1;
        if(yPos == 5)
        {
            yPos = 0
        }
        return yPos;
    }

}

File: gatorlight.ts
/**
* Andy England @ SparkFun Electronics
* September 6, 2018
* https://github.com/sparkfun/pxt-light-bit

* Development environment specifics:
* Written in Microsoft Makecode
* Tested with a SparkFun gatorlight sensor and micro:bit
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


/**
 * Functions to operate the gatorlight sensor
 */


//% color=#f44242 icon="\uf185"
namespace gatorlight {

    // Functions for reading light from the gatorlight in lux or straight adv value

    /**
    * Reads the number
    */
    //% weight=30 blockId="xPos" block="Get LED xPos"
    export function xPos(): number{
      
        return getXPos()
      }
    
    //% weight=30 blockId="yPos" block="Get LED yPos"
    export function yPos(): number {

        return getYPos()
    }

	/**
     * Function used for simulator, actual implementation is in gatorlight.cpp
     */
    //% shim=gatorlight::getXPos
    function getXPos(): number {
        // Fake function for simulator
        return 0
    }
    /**
     * Function used for simulator, actual implementation is in gatorlight.cpp
     */
    //% shim=gatorlight::getYPos
    function getYPos(): number {
        // Fake function for simulator
        return 0
    }

    }
